packages/flutter_tools/lib/src/test/runner.dart

```diff
@@ -102,8 +102,10 @@ class _FlutterTestRunnerImpl implements FlutterTestRunner {
         '--pause-after-load',
       if (machine)
         ...<String>['-r', 'json']
-      else
-        ...<String>['-r', reporter ?? 'compact'],
+      else if (globals.platform.environment['GITHUB_ACTIONS']?.toLowerCase() == 'true')
```

Comment:
The test framework defaults to github actions on github, you shouldn't need to set this

Reply:
I was trying to move your logic for assigning `github` as the reporter downstream from where you put it.

https://github.com/flutter/flutter/pull/115137/files#diff-328d8a42e42b84152d577569aa043fdfa16977190165f1a11897f6f47d81c1a5

Is this not relevant?

Reply:
if you're removing the default, the test package's default logic for selecting the reporter should work, right?

Reply:
Yes, I believe so. I found this [snippet](https://github.com/dart-lang/test/blob/master/pkgs/test_core/lib/src/util/io.dart#L90-L94) here within the repo for `package:test_core` which seems to be doing the same thing.

```dart
/// Detect whether we're running in a Github Actions context.
///
/// See
/// https://docs.github.com/en/actions/learn-github-actions/environment-variables.
bool get inGithubContext => Platform.environment['GITHUB_ACTIONS'] == 'true';
```

Seems like it gets used in the `defaultReporter` variable

```dart
final defaultReporter = inTestTests
    ? 'expanded'
    : inGithubContext
        ? 'github'
        : canUseSpecialChars
            ? 'compact'
            : 'expanded';
```
I'll go ahead and remove that conditional